pkg/manifest/os.go

```diff
@@ -853,13 +857,55 @@ func (p *OS) serialize() osbuild.Pipeline {
 	return pipeline
 }
 
-func prependKernelCmdlineStage(pipeline osbuild.Pipeline, kernelOptions string, pt *disk.PartitionTable) osbuild.Pipeline {
+func prependKernelCmdlineStage(pipeline osbuild.Pipeline, kernelOptions []string, pt *disk.PartitionTable, mountUnits bool) osbuild.Pipeline {
 	rootFs := pt.FindMountable("/")
 	if rootFs == nil {
 		panic("root filesystem must be defined for kernel-cmdline stage, this is a programming error")
 	}
+
+	// if /usr is on a separate filesystem, it needs to be defined in the
+	// kernel cmdline options for autodiscovery (when there's no /etc/fstab)
+	// see:
+	//  - https://github.com/systemd/systemd/issues/24027
+	//  - https://github.com/systemd/systemd/pull/33397
+	if usrFs := pt.FindMountable("/usr"); usrFs != nil && mountUnits {
```

Comment:
It slightly easier to me to combine the `if mountUnits` here and on line 884 into one block with a nested `FindMountable` in it.

Reply:
I considered it but I felt the separation made it easier to reason with.  I'm not against changing it though.